### PR TITLE
#458 Create geolocation api interface

### LINKED ISSUE #458: GPS localization interface
Try to get GPS localization while working app - it will be using to collect points.
Prepare interface for GPS localization - make it easier to use in code.

https://developer.mozilla.org/pl/docs/Web/API/Geolocation_API

### PR DESCRIPTION
**Pull request by Bedpaw**
Close #458

### CODE REVIEW

File: src/utils/geolocation.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    api.getCurrentPosition(successCallback, errorCallback, options);
+
+  const getCurrentPositionPromise = (options = positionOptions) =>
+    new Promise((resolve, reject) =>
+      api.getCurrentPosition(resolve, reject, options));
+
+  const isEnabled = (options = positionOptions) =>
+    getCurrentPositionPromise(options)
+      .then(true)
+      .catch(false);
```
Comment (dbetka) [inline src/utils/geolocation.js:45]: Why `isEnabled` method return promise?
And why I get current position on `then` for code like below:
```
geolocationApi().isEnabled().then(value => console.log(value)) 
// console output: GeolocationPosition {coords: GeolocationCoordinates, timestamp: 1626626143515}
```

File: src/utils/geolocation.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+const positionOptions = {
+  /* Default options from docs */
+  enableHighAccuracy: true,
+  timeout: 0xFFFFFFFF,
+  maximumAge: 0,
+};
+
+export const geolocationApi = () => {
```
Comment (dbetka) [inline src/utils/geolocation.js:8]: `geolocationApi` function has appearance and few features typical for `class`. I need explanation why it's not write in class - we can talk about it.

File: src/utils/geolocation.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+const positionOptions = {
+  /* Default options from docs */
+  enableHighAccuracy: true,
+  timeout: 0xFFFFFFFF,
```
Comment (dbetka) [inline src/utils/geolocation.js:4]: Why timeout is equal `0xFFFFFFFF`?
What unit is using for `maximumAge`?

**Review by dbetka — CHANGES_REQUESTED** [review]
Answer and fix @Bedpaw

**Comment by Bedpaw** [comment]
I will move this code to wdk repo and fix this issues there -> then I will close this pr

**Pull request closed without merging**